### PR TITLE
Reference module with `.c`

### LINKED ISSUE #13657: Access restriction to data and config folder
Problem: 

The data and the config folder are protected via an .htaccess file. The change from Apache 2.2 to 2.4 causes a refactoring of authorization process. To capture this a new template for .htaccess was delivered with ownclud.

### Expected behaviour
If I try to get access via the browser

www.myowncloud.com/data/owncloud.db

I would expect two things:

In the browser: "Forbidden"
In the LogFile: ... [authz_core .... client denied by server configuration: ... data/owncloud.db

### Actual behaviour

In the browser: "Forbidden"
In the LogFile: ... [access_compat .... client denied by server configuration: ... data/owncloud.db

This means that not the "Require all denied" is used but the "Deny from all". This can be seen in the module (mod_access_compat) capturing htaccess statement. 

In case mod_access_compat is not loaded I think it can lead to a security problem.

### Server configuration
Apache 2.4
Owncloud 7.0.4

## Solution:

I investigated a bit and found out that within the IfModule Statement the .c is missing. It should be written:
```
<ifModule mod_authz_core.c>
Require all denied
</ifModule>
```

instead of 

```
<ifModule mod_authz_core>
Require all denied
</ifModule>
```

Can you agree?

### PR DESCRIPTION
**Pull request by LukasReschke**
Fixes https://github.com/owncloud/core/issues/13657

@DeepDiver1975 @nickvergessen

### CODE REVIEW

**Comment by LukasReschke** [comment]
Remark: We should run the `updateDataDirectory` routine also on updates – will file a different issue about that.

File: .htaccess [hunk 1]
```diff
@@ -13,7 +13,7 @@ php_value post_max_size 513M
 php_value memory_limit 512M
 php_value mbstring.func_overload 0
 php_value always_populate_raw_post_data -1
-<IfModule env_module>
+<IfModule env_module.c>
```
Comment (DeepDiver1975) [inline .htaccess]: @LukasReschke sure about this change? This would mean it did never work - which is not the case - right?

Comment (LukasReschke) [inline .htaccess]: Switched to `mod_env.c` which should always work that way

**Comment by Xenopathic** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Still works. :+1: Also there is no warning in the admin backend

**Pull request merged**